src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectDataInterner.cs

```diff
@@ -13,7 +13,7 @@
 
 namespace ILCompiler
 {
-    public sealed partial class ObjectDataInterner
+    public partial class ObjectDataInterner
```

Comment:
Changes here can be reverted assuming the removal of `CanFoldAlwaysObjectDataInterner`.